projects/plugins/jetpack/extensions/blocks/videopress/resumable-upload/use-uploader.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * External dependencies
+ */
+import * as tus from 'tus-js-client';
+
+export const getJWT = function ( key ) {
+	return new Promise( function ( resolve, reject ) {
+		const extras = key ? { data: { key } } : {};
+		// eslint-disable-next-line no-undef
+		wp.media
+			.ajax( 'videopress-get-upload-jwt', { async: true, ...extras } )
+			.done( function ( response ) {
+				resolve( {
+					token: response.upload_token,
+					blogId: response.upload_blog_id,
+					url: response.upload_action_url,
+				} );
+			} )
+			.fail( function ( reason ) {
+				reject( reason );
+			} );
+	} );
+};
+
+const jwtsForKeys = {};
+
+export const resumableUploader = ( { onError, onProgress, onSuccess } ) => {
+	return ( file, data ) => {
+		const upload = new tus.Upload( file, {
+			onError: onError,
+			onProgress: onProgress,
+			endpoint: data.url,
+			resume: true,
+			removeFingerprintOnSuccess: true,
+			withCredentials: false,
+			autoRetry: true,
+			overridePatchMethod: false,
+			chunkSize: 500000, // 500 Kb.
+			allowedFileTypes: [ 'video/*' ],
+			metadata: {
+				filename: file.name,
+				filetype: file.type,
+			},
+			retryDelays: [ 0, 1000, 3000, 5000, 10000 ],
+			onAfterResponse: function ( req, res ) {
+				// Why is this not showing the x-headers?
+				if ( res.getStatus() >= 400 ) {
+					return;
+				}
+
+				const GUID_HEADER = 'x-videopress-upload-guid';
+				const guid = res.getHeader( GUID_HEADER );
+				if ( guid ) {
+					onSuccess && onSuccess( guid );
+					return;
+				}
+
+				const headerMap = {
+					'x-videopress-upload-key-token': 'token',
+					'x-videopress-upload-key': 'key',
+					'x-videopress-upload-blog-id': 'blogId',
```

Comment:
nit: not a huge deal, but it doesn't appear this is used here

Reply:
@pgk shall I just remove that line?

Reply:
I went ahead and removed it in de38376